src/components/GitHubSync.tsx

```diff
@@ -0,0 +1,947 @@
+                  "_blank",
+                  `width=${WINDOW_WIDTH}, height=${WINDOW_HEIGHT}, top=${WINDOW_TOP}, left=${WINDOW_LEFT}`
+                );
+
+                let attemptCount = 0;
+                const check = () => {
+                  if (attemptCount < 10) {
+                    apiPost({
+                      path: "access-token",
+                      domain: isDev
+                        ? "https://api.samepage.ngrok.io"
+                        : "https://api.samepage.network",
+                      data: { state },
+                    }).then((r) => {
+                      if (r.accessToken) {
+                        localStorageSet("oauth-github", r.accessToken);
+                        setGitHubAccessToken(r.accessToken);
+                        setClickedInstall(false);
+                        authWindow.current?.close();
+                      } else {
+                        attemptCount++;
+                        setTimeout(check, 1000);
+                      }
+                    });
+                  } else {
+                    setError("Something went wrong.  Please contact support.");
+                  }
+                };
+                setTimeout(check, 1500);
+              }}
+            />
+          )}
+
+          {repoSelectEnabled && (
+            <Tag intent="success" large className="text-center w-52">
+              Authorized
+            </Tag>
+          )}
+          <h3>GitHub Sync Settings</h3>
+          <a
+            onClick={() => {
+              window.roamAlphaAPI.ui.mainWindow.openPage({
+                page: { title: CONFIG_PAGE },
+              });
+              setIsOpen(false);
+            }}
+          >
+            {`[[${CONFIG_PAGE}]]`}
+          </a>
+        </div>
+      </div>
+      <div className={Classes.DIALOG_FOOTER}>
+        <div className={Classes.DIALOG_FOOTER_ACTIONS}>
+          <span style={{ color: "darkred" }}>{error}</span>
+        </div>
+      </div>
+    </Dialog>
+  );
+};
+
+// Init
+const initializeGitHubSync = async (args: OnloadArgs) => {
+  const unloads = new Set<() => void>();
+  const toggle = async (flag: boolean) => {
+    if (flag && !enabled) {
+      const { observer } = await createConfigObserver({
+        title: "roam/js/github-sync",
+        config: {
+          tabs: [
+            {
+              id: "home",
+              fields: [
+                {
+                  // @ts-ignore
+                  Panel: SelectPanel,
+                  title: "Node Select",
+                  description: "Select a node to pull comments from",
+                  options: {
+                    items: [
+                      "None",
+                      ...getDiscourseNodes()
+                        .map((node) => node.text)
+                        .filter((text) => text !== "Block"),
+                    ],
+                  },
+                  defaultValue: "None",
```

Comment:
maybe this should create be a CustomPanel and create a block on select/enter so the user can have multiple defined?